### ISSUE TITLE
*Datatable* - _(Datatable)_ - Changes not Saving back to Record in Record Collection

### ISSUE THREAD

**Issue by kmoser13** [alexed1/LightningFlowComponents#1373]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

### Steps to reproduce

1. Add Datatable component to a screen flow
2. Make a field editable
3. Debug the flow
4. Make changes to the editable values in the datatable
5. Click the Save button (nothing seems to happen when looking at the flow debug information)
6. Click Next on the flow screen
7. See that the changed values were not saved back to the record collection (see [this video](https://app.screencast.com/VsMXBupiuQvc3?conversation=pkkkEJx3irZ9mgOkQyca0n) of the issue for more info)

### Expected behaviour
Values that are changed, should be saved back to the record within the record collection

### Actual behaviour
Values that are changed are not saved back to the record within the collection

### Screenshots
See [this video](https://app.screencast.com/VsMXBupiuQvc3?conversation=pkkkEJx3irZ9mgOkQyca0n) of the issue for more info

### Debug Logs
If applicable, please insert debug logs here
<details>
<summary>Debug Log Information</summary>

```
Paste the output of the logs here.
```
How the Interview Started
System Admin (0051K000008ffmf) started the flow interview.
API Version for Running the Flow: 57
GET RECORDS: Opportunities
Find all Opportunity records.
Store the values of these currently referenced fields in Opportunities: Amount, Id, CloseDate
Because Opportunities is passed to an action, subflow, or Lightning component, store the values of all Opportunity fields that the running user has access to.
Result
Successfully found records.
Transaction Committed
Any records that the flow was ready to create, update, or delete were committed to the database.
SCREEN: test
Lightning Component: OpportunityUpdate
Screen component: c:datatable
Inputs:
cb_matchCaseOnFilters = (CB_FALSE)
columnEdits = (Amount:true, CloseDate:true)
columnFields = (Name,Amount,CloseDate)
columnWidths = (Name:445, Amount:160, CloseDate:130)
objectName = (Opportunity)
tableData = Opportunities ([{AccountId=0015500001YwWdxAAF, Account_Has_CSM__c=false, Account_S5_Notification__c=Test, Adjusted_Revenue__c=0.0, Amount=100000.0, Archive_CRM_Account_ID__c=null, Archive_CRM_Activity_ID__c=null, Archive_CRM_Contact_ID__c=null, Archive_CRM_Opportunity_ID__c=null, Archive_CRM_Owner_ID__c=null, Audit_Close_Date__c=true, Automation_Bypass_Date__c=null, Automation_Bypass__c=false, BD_Notes__c=null, Budget_Confirmed__c=false, COLA_Index_Increase__c=null, CSM_Override__c=false, CampaignId=null, Capabilities_Score__c=0.0, Capabilities__c=null, Category__c=null, Client_Info_Complete__c=false, Client_Success_Manager__c=null, Cloned_Opportunity__c=null, CloseDate=2023-03-20 00:00:00, Close_Date_Alert__c= , Close_Other_Reason__c=null, Close_Reason__c=null, Closed_Won_Reason__c=null, Closed_Won_Revenue_Adjusted__c=0.0, Closed_Won_Revenue_In_Year__c=0.0, Closed_Won_Revenue__c=0.0, Communication_Cadence__c=null, Competencies_Score__c=0.0, Competencies__c=null, Confidence_of_Win__c=null, ContactId=0035500001NVKF6AAP, ContractId=null, Contract_End_Date__c=null, Contract_Length__c=null, Contract_Start__c=2023-05-01 00:00:00, Contract_Term__c=3.0, Contract_URL__c=null, Contract_Upload_Date__c=null, Contract_Uploaded__c=false, CreatedById=0051K000008ffmfQAA, CreatedDate=2023-03-20 18:53:32, Current_Go_Get_Adjusted__c=0.0, Current_Go_Get_Estimated__c=0.0, Current_In_Year_Adjusted__c=0.0, Current_In_Year_Estimated__c=0.0, Current_State__c=null, Current_Year_Revenue__c=0.0, Current_Yr_Rev_roll_up__c=0.0, Data_Room_Access__c=false, DealHub_Quote_Id__c=null, DealHub_Status__c=null, DealRoom_Status__c=null, DealRoom_URL__c=null, Decision_Date__c=null, Decision_Maker__c=null, Deferred_Revenue__c=0.0, Description=null, Description_Entered__c=false, Discovery_Completed__c=false, Display_S5_Email_Message__c=false, Domain__c=https://icario--dev--c.sandbox.vf.force.com/, EBITDA__c=null, Earliest_Planned_Launch_Date__c=2023-05-01 00:00:00, Entrance_into_Data_Room__c=null, Estimated_Revenue__c=null, Executive_Sponsor__c=null, ExpectedRevenue=1000.0, Fees_At_Risk__c=null, Fiscal=2023 1, FiscalQuarter=1, FiscalYear=2023, ForecastCategory=Pipeline, ForecastCategoryName=Pipeline, Go_Live_Date__c=2023-05-01 00:00:00, HCA_Type__c=null, HasOpenActivity=false, HasOpportunityLineItem=true, HasOverdueTask=false, Has_Contact__c=true, Health_Care_Engagement_Strategist__c=0051K000008ffmfQAA, Hubspot_Created_Date__c=null, Hubspot_Deal_Name__c=null, Hubspot_ID_Oppty__c=null, INBAT_Score__c=null, IXG_Email_Address__c=ixg@icariohealth.com, IXG_Team_Notes__c=null, Id=0065500000LndXHAAZ, In_Year_Adjusted__c=0.0, In_Year_Estimated__c=0.0, In_Year_Revenue__c=0.0, In_Yr_Rev_roll_up__c=0.0, Include_in_Week_Ended__c=true, Influencer__c=null, IsClosed=false, IsDeleted=false, IsWon=false, Is_Test_Opportunity__c=false, LastActivityDate=null, LastAmountChangedHistoryId=0085500000TMwHvAAL, LastCloseDateChangedHistoryId=null, LastModifiedById=0051K000008ffmfQAA, LastModifiedDate=2023-03-21 17:57:56, LastReferencedDate=2023-03-22 17:34:26, LastStageChangeDate=null, LastViewedDate=2023-03-22 17:34:26, Last_Activity_Date__c=null, Last_Contacted__c=null, LeadSource=Unknown, Lead_Source_Detail_Oppty__c=null, Line_of_Business_SDoc__c=Medicaid, Line_of_Business__c=Medicaid, Link_to_Data_Room__c=null, Loss_Reason_Note__c=null, Loss_Reason__c=null, Lost_Revenue_Adjusted__c=0.0, Lost_Revenue_In_Year__c=0.0, Lost_Revenue__c=0.0, Master_PID__c=00001, Multiple_LOBs__c=null, My_Opportunity__c=true, Name=TST_Medicaid_2023, New_Markets__c=null, New_Revenue_Adjusted__c=0.0, New_Revenue_In_Year__c=0.0, New_Revenue__c=0.0, NextStep=null, Next_In_Year_Adjusted__c=0.0, Next_In_Year_Estimated__c=0.0, Non_Standard_HCAs__c=null, On_site_Date__c=null, On_site_Notes__c=null, Opp_Owner_Name__c=System Admin, Opportunity_ID_18__c=0065500000LndXHAAZ, Opportunity_Name__c=TST_Medicaid_2023, Opportunity_Priority__c=null, Opportunity_Sync__c=2023-03-20 19:09:23, Opportunity__c=1.0, Original_Source_Data_1_Oppty__c=null, Original_Source_Type_Oppty__c=null, Other_Lead_Source__c=null, Outcome__c=null, Outstanding_Requirements_S2__c=- Enter a Description
, Outstanding_Requirements_S3__c=- Upload Proposal to Opportunity or Create Quote in DealHub
, Outstanding_Requirements_S5__c=- Enter Missing PRD Info (click Add Missing PRD Info button to review), Outstanding_Requirements_Won__c=- Upload Contract to Opportunity
, OwnerId=0051K000008ffmfQAA, Owner_Assigned_Date__c=null, PRD_Count__c=1.0, PRD_Info_Missing_Closed_Won__c=0.0, PRD_Info_Missing_S5__c=1.0, Partnership_Potential__c=null, Potential_M_A_Target__c=false, Pricebook2Id=01s410000025AymAAE, Pricing_Model_Detail__c=null, Pricing_Model__c=null, Pricing_Notes__c=null, Primary_Margin_Percent__c=null, Prior_Week_Ending__c=2023-03-10 00:00:00, Prior_Years_Revenue__c=null, Priority_Improvements_to_help_Close_Rene__c=null, Probability=1.0, Product_OLD__c=null, Professional_Services_Only__c=false, Program_Objectives_Other__c=null, Program_Year__c=2023, Program_from_Hubspot__c=null, Progression_Revenue_Adjusted__c=0.0, Progression_Revenue_In_Year__c=0.0, Progression_Revenue__c=0.0, Proposal_Upload_Date__c=null, Proposal_Uploaded__c=false, PushCount=0, ROI_Analysis_Completed__c=false, RecordTypeId=0121K000000zLMAQA2, Referral_Source__c=null, Regression_Revenue_Adjusted__c=0.0, Regression_Revenue_In_Year__c=0.0, Regression_Revenue__c=0.0, Related_Opportunity__c=null, Renewal_Complexity__c=null, Renewal_Notes__c=null, Requirement_Start_Date__c=2020-12-31 00:00:00, Revenue__c=100000.0, Rewards_Estimate__c=null, Risks__c=null, S5_Sent_Date__c=null, Sales_Leader_Notes__c=null, Sender_S5_Notification__c=System Admin, Signed_NDA__c=null, Single_Sign_On__c=No, Skip_HCES_Survey__c=false, Snapshot_Est_Revenue__c=null, Source_Key__c=null, StageName=S0 - Suspect/Target, Strategy_Classification__c=null, SyncedQuoteId=null, SystemModstamp=2023-03-21 17:57:56, Target_Rationale__c=null, Target_Scoring__c=null, Targeted_Members__c=null, Term_for_Convenience__c=false, Top_Renewal_Opportunity__c=false, Total_Contract_Value__c=0.0, Total_Program_Rev_roll_up__c=0.0, Trend_Group__c=2023S0 - Suspect/Target|New Business}System Admin, Type=New Business, Under_Contract__c=false, Valuation_Range__c=null, Week_Ended_Date_Prior__c=null, Week_Ended_Date__c=null, Week_Ended_Est_Revenue_Prior__c=null, Week_Ended_Go_Live_Date_Prior__c=null, Week_Ended_Go_Live_Date__c=null, Week_Ended_In_Year_Est_Rev__c=null, Week_Ended_In_Year_Est_Revenue_Prior__c=null, Week_Ended_Probability_Prior__c=null, Week_Ended_Probability__c=null, Week_Ended_Stage_Prior__c=null, Week_Ended_Stage_Progression__c=No Change, Week_Ended_Stage__c=null, Week_Ending__c=2023-03-17 00:00:00, Win_Loss_Interview_Recording__c=null, Win_Loss_Interview_Status__c=Not Started, Year_1_Program_End_Date__c=null}])
tableHeight = (400px)
Outputs:
numberOfRowsEdited = 1
numberOfRowsSelected = 0
objectName = Opportunity
outputEditedRows = [[Opportunity (Audit_Close_Date__c:true, Potential_M_A_Target__c:false, LastModifiedDate:Tue Mar 21 17:57:56 GMT 2023, Priority_Improvements_to_help_Close_Rene__c:null, Loss_Reason_Note__c:null, Current_Yr_Rev_roll_up__c:0, EBITDA__c:null, Opportunity_ID_18__c:0065500000LndXHAAZ, Requirement_Start_Date__c:Thu Dec 31 00:00:00 GMT 2020, Outstanding_Requirements_S2__c:- Enter a Description
, HasOpenActivity:false, New_Revenue__c:0, Proposal_Uploaded__c:false, Lead_Source_Detail_Oppty__c:null, Estimated_Revenue__c:null, Week_Ended_Go_Live_Date__c:null, Related_Opportunity__c:null, IsDeleted:false, Week_Ended_In_Year_Est_Rev__c:null, Health_Care_Engagement_Strategist__c:0051K000008ffmfQAA, Week_Ended_Stage__c:null, Week_Ended_In_Year_Est_Revenue_Prior__c:null, Regression_Revenue_Adjusted__c:0, Archive_CRM_Account_ID__c:null, Program_from_Hubspot__c:null, Progression_Revenue_In_Year__c:0, COLA_Index_Increase__c:null, Capabilities_Score__c:0, IsClosed:false, Skip_HCES_Survey__c:false, Pricing_Model_Detail__c:null, New_Revenue_Adjusted__c:0, CampaignId:null, Current_Go_Get_Estimated__c:0, Opportunity_Name__c:TST_Medicaid_2023, Closed_Won_Reason__c:null, RecordTypeId:0121K000000zLMAQA2, Original_Source_Type_Oppty__c:null, StageName:S0 - Suspect/Target, Decision_Maker__c:null, Probability:1, Week_Ended_Stage_Progression__c:No Change, Under_Contract__c:false, On_site_Date__c:null, SystemModstamp:Tue Mar 21 17:57:56 GMT 2023, Automation_Bypass_Date__c:null, IXG_Email_Address__c:ixg@icariohealth.com, Original_Source_Data_1_Oppty__c:null, Risks__c:null, Prior_Week_Ending__c:Fri Mar 10 00:00:00 GMT 2023, Contract_Term__c:3, Revenue__c:100000, DealHub_Status__c:null, Closed_Won_Revenue__c:0, Current_In_Year_Estimated__c:0, AccountId:0015500001YwWdxAAF, New_Markets__c:null, Lost_Revenue_In_Year__c:0, Outcome__c:null, Owner_Assigned_Date__c:null, Master_PID__c:00001, Name:TST_Medicaid_2023, Week_Ended_Go_Live_Date_Prior__c:null, Entrance_into_Data_Room__c:null, Partnership_Potential__c:null, CreatedById:0051K000008ffmfQAA, PRD_Info_Missing_S5__c:1, Current_State__c:null, Outstanding_Requirements_S3__c:- Upload Proposal to Opportunity or Create Quote in DealHub
, Archive_CRM_Opportunity_ID__c:null, DealRoom_URL__c:null, Pricing_Model__c:null, Contract_Length__c:null, Description_Entered__c:false, LastViewedDate:Wed Mar 22 17:34:26 GMT 2023, Referral_Source__c:null, Capabilities__c:null, Fiscal:2023 1, Week_Ended_Stage_Prior__c:null, My_Opportunity__c:true, Sales_Leader_Notes__c:null, CreatedDate:Mon Mar 20 18:53:32 GMT 2023, HasOverdueTask:false, Discovery_Completed__c:false, NextStep:null, Hubspot_ID_Oppty__c:null, Display_S5_Email_Message__c:false, LastReferencedDate:Wed Mar 22 17:34:26 GMT 2023, Program_Objectives_Other__c:null, Decision_Date__c:null, Outstanding_Requirements_Won__c:- Upload Contract to Opportunity
, Primary_Margin_Percent__c:null, IsWon:false, Line_of_Business__c:Medicaid, Trend_Group__c:2023S0 - Suspect/Target|New Business}System Admin, Lost_Revenue__c:0, Other_Lead_Source__c:null, Sender_S5_Notification__c:System Admin, Last_Activity_Date__c:null, Automation_Bypass__c:false, Next_In_Year_Estimated__c:0, Client_Success_Manager__c:null, Contract_Start__c:Mon May 01 00:00:00 GMT 2023, Snapshot_Est_Revenue__c:null, Next_In_Year_Adjusted__c:0, Signed_NDA__c:null, Current_In_Year_Adjusted__c:0, Progression_Revenue__c:0, Account_S5_Notification__c:Test, Hubspot_Deal_Name__c:null, PRD_Count__c:1, Opportunity_Priority__c:null, Opportunity__c:1, Renewal_Complexity__c:null, CloseDate:Sun Dec 31 00:00:00 GMT 2023, Targeted_Members__c:null, In_Year_Adjusted__c:0, Line_of_Business_SDoc__c:Medicaid, Week_Ended_Est_Revenue_Prior__c:null, Budget_Confirmed__c:false, Hubspot_Created_Date__c:null, Rewards_Estimate__c:null, Competencies__c:null, Target_Scoring__c:null, Executive_Sponsor__c:null, FiscalQuarter:1, Win_Loss_Interview_Recording__c:null, Non_Standard_HCAs__c:null, Cloned_Opportunity__c:null, Contract_End_Date__c:null, ROI_Analysis_Completed__c:false, Product_OLD__c:null, Contract_URL__c:null, Opp_Owner_Name__c:System Admin, Lost_Revenue_Adjusted__c:0, Communication_Cadence__c:null, Regression_Revenue_In_Year__c:0, Confidence_of_Win__c:null, Id:0065500000LndXHAAZ, Go_Live_Date__c:Mon May 01 00:00:00 GMT 2023, LastCloseDateChangedHistoryId:null, Renewal_Notes__c:null, Description:null, Data_Room_Access__c:false, Domain__c:https://icario--dev--c.sandbox.vf.force.com/, Target_Rationale__c:null, Contract_Uploaded__c:false, Regression_Revenue__c:0, Multiple_LOBs__c:null, LastAmountChangedHistoryId:0085500000TMwHvAAL, Total_Contract_Value__c:0, DealRoom_Status__c:null, OwnerId:0051K000008ffmfQAA, Source_Key__c:null, Contract_Upload_Date__c:null, SyncedQuoteId:null, Year_1_Program_End_Date__c:null, Close_Reason__c:null, Link_to_Data_Room__c:null, New_Revenue_In_Year__c:0, Professional_Services_Only__c:false, Loss_Reason__c:null, Top_Renewal_Opportunity__c:false, Amount:150000, Win_Loss_Interview_Status__c:Not Started, Closed_Won_Revenue_Adjusted__c:0, In_Year_Estimated__c:0, Current_Year_Revenue__c:0, LastStageChangeDate:null, Type:New Business, CSM_Override__c:false, Earliest_Planned_Launch_Date__c:Mon May 01 00:00:00 GMT 2023, Total_Program_Rev_roll_up__c:0, LastActivityDate:null, Archive_CRM_Contact_ID__c:null, Fees_At_Risk__c:null, Close_Date_Alert__c: , On_site_Notes__c:null, HasOpportunityLineItem:true, Is_Test_Opportunity__c:false, Week_Ended_Date_Prior__c:null, ForecastCategory:Pipeline, HCA_Type__c:null, Valuation_Range__c:null, BD_Notes__c:null, Pricing_Notes__c:null, Competencies_Score__c:0, Closed_Won_Revenue_In_Year__c:0, Influencer__c:null, S5_Sent_Date__c:null, Close_Other_Reason__c:null, Has_Contact__c:true, Deferred_Revenue__c:0, In_Year_Revenue__c:0, LeadSource:Unknown, Category__c:null, Pricebook2Id:01s410000025AymAAE, ForecastCategoryName:Pipeline, FiscalYear:2023, Archive_CRM_Activity_ID__c:null, Include_in_Week_Ended__c:true, PRD_Info_Missing_Closed_Won__c:0, Program_Year__c:2023, ContactId:0035500001NVKF6AAP, Last_Contacted__c:null, Week_Ended_Date__c:null, Single_Sign_On__c:No, Current_Go_Get_Adjusted__c:0, ExpectedRevenue:1000, Account_Has_CSM__c:false, Client_Info_Complete__c:false, Progression_Revenue_Adjusted__c:0, Prior_Years_Revenue__c:null, Week_Ended_Probability__c:null, PushCount:0, In_Yr_Rev_roll_up__c:0, Term_for_Convenience__c:false, Archive_CRM_Owner_ID__c:null, Adjusted_Revenue__c:0, Opportunity_Sync__c:Mon Mar 20 19:09:23 GMT 2023, DealHub_Quote_Id__c:null, ContractId:null, Outstanding_Requirements_S5__c:- Enter Missing PRD Info (click Add Missing PRD Info button to review), INBAT_Score__c:null, Strategy_Classification__c:null, IXG_Team_Notes__c:null, LastModifiedById:0051K000008ffmfQAA, Week_Ended_Probability_Prior__c:null, Week_Ending__c:Fri Mar 17 00:00:00 GMT 2023, Proposal_Upload_Date__c:null)]]
outputEditedRowsString =
outputEditedSerializedRows =
outputSelectedRow = null
outputSelectedRows = []
outputSelectedRowsString =
selectedRowKeyValue =
sortDirection =
sortedBy =
wizColumnFields = Name,Amount,CloseDate
wizColumnWidths = Name:445, Amount:160, CloseDate:130
wizSObject = Opportunity

Selected Navigation Button: NEXT
ASSIGNMENT: Assign to Variable
{!OpportunityList} Equals {!Opportunities}
Result
{!OpportunityList} = "[Opportunity (0065500000LndXHAAZ)]"
LOOP: Loop
Loop Through: [0065500000LndXHAAZ]
Iteration: 0
Current iteration item: 0065500000LndXHAAZ
Transaction Committed
Any records that the flow was ready to create, update, or delete were committed to the database.

**Comment by dalton-compass** [comment]
@kmoser13 what did the solution end up being?  I am having the same issue trying to display a record collection and make inline edits.  The changes won't persist when trying to create the records after the Datatable screen.

**Comment by ericrsmith35** [comment]
How do you have the datatable and the Create Records configured?

**Comment by kmoser13** [comment]
Hi guys!  This was over 2 years ago, so I'm not sure what ended up being built at that time. I'm guessing the version of this datatable component has been updated since then as well. Sorry I'm not more help!

**Comment by dalton-compass** [comment]
@ericrsmith35 I figured it out, it was user error.  I was passing in the Record Collection Variable and then trying to access that same collection again.  I found that I must use either the "Output Remaining Rows" or "Output Edited Rows" options from the DataTable component later in my flow, instead of the original dataset.  Just a misunderstanding on my part on how/where data is being written to.

@kmoser13, sorry to bother, I realized the date after commenting.  Thank you, though!

Putting all this here for the next person who runs into this issue.
